irohad/consensus/yac/storage/yac_common.hpp

```diff
@@ -29,9 +29,9 @@ namespace iroha {
 
       struct VoteMessage;
 
-      using ProposalHash = decltype(YacHash::proposal_hash);
+      using ProposalHash = decltype(YacHash::vote_hashes_.proposal_hash);
```

Comment:
Imo it's better define types here and then reuse them in YacHash definition. With that approach ctor types can be also be "generic"